hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieRealtimeInputFormatUtils.java

```diff
@@ -82,7 +82,7 @@
       HoodieTableMetaClient metaClient = partitionsToMetaClient.get(partitionPath);
       if (!fsCache.containsKey(metaClient)) {
 
-        HoodieTableFileSystemView fsView = FileSystemViewManager.createInMemoryFileSystemView(metaClient,
+        HoodieTableFileSystemView fsView = FileSystemViewManager.createInMemoryFileSystemViewForReaders(metaClient,
```

Comment:
Can we avoid special naming methods based on caller. We should keep stuff like `FileSystemViewManager` totally agnostic of reader/writer. If you want special behavior, can you please add a new method arg and control it like that.

Reply:
Will do.